### ISSUE TITLE
Fix #882 has created new problem

### ISSUE THREAD

**Issue by refdoc** [crosswire/xiphos#885]
As result of #882 the crashes are now avoided but both the scroll down menu for book names and the chapter choser are misplaced 

(Debian stretch)

**Comment by karlkleinpaste** [comment]
the entirety of the change for #882 is to replace "0,0" pairs in menu popups with GDK gravity indicators, e.g.:
-               gtk_menu_popup_at_widget(GTK_MENU(menu), widget, 0, 0, NULL);
+               gtk_menu_popup_at_widget(GTK_MENU(menu), widget, GDK_GRAVITY_SOUTH_WEST, GDK_GRAVITY_NORTH_WEST, NULL);
 
how can this possibly have had the slightest effect on chooser placement? are we looking at an actual xiphos problem, or instead a GTK or Wayland problem?

**Comment by karlkleinpaste** [comment]
i hate the github editor. i hate it, hate it, hate it.
those 2 lines were the '-'/'+' lines of "git diff" of one example use of gtk_menu_popup_at_widget.

**Comment by greg-hellings** [comment]
@karlkleinpaste I've updated your comment to properly show and highlight that as a patch diff.

**Comment by refdoc** [comment]
Ok some more info re what is happening:

The three drop down menus (book name, chapter no, verse no) will, if too long move to the left edge of the Xiphos window. So, if a chapter e.g. has only 10 verses the drop down menu will be in the usual and normal place. If a book has only 10 chapters, same. But if a book has a lot of chapters or a chapter lots of verses then the menu will detach and hang on the edge of Xiphos' window.

**Comment by LAfricain** [comment]
@karlkleinpaste  @greg-hellings @refdoc Is this bug still an issue? I've never encountered it on my Ubuntu systems. By the way, we've moved the chapter and verse display to a column layout—could that be why the problem has been resolved?

**Comment by karlkleinpaste** [comment]
I agree, this seems to be no longer a problem, if it ever was. (I never saw it happen.) Closing because, at the least, 2D BCV selectors make things shorter anyway.
